### PR TITLE
ipfs-cluster: 1.0.1 -> 1.0.2

### PR DESCRIPTION
**Pull request by Luflosi**
###### Description of changes
https://github.com/ipfs/ipfs-cluster/releases/tag/v1.0.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by applePrincess — APPROVED** [review]

**Comment by applePrincess** [comment]
```
[nix-shell:~/.cache/nixpkgs-review/pr-181382]$ 2022-07-19T17:18:03.624Z	INFO	service	ipfs-cluster-service/daemon.go:49	Initializing. For verbose output run with "-l debug". Please wait...
2022-07-19T17:18:03.626Z	INFO	badger	badger@v1.6.2/logger.go:46	All 0 tables opened in 0s

2022-07-19T17:18:03.760Z	INFO	badger	badger@v1.6.2/logger.go:46	Replaying file id: 0 at offset: 0

2022-07-19T17:18:03.760Z	INFO	badger	badger@v1.6.2/logger.go:46	Replay took: 1.416µs

2022-07-19T17:18:03.760Z	INFO	service	ipfs-cluster-service/daemon.go:267	Datastore backend: badger
2022-07-19T17:18:03.767Z	INFO	cluster	ipfs-cluster/cluster.go:137	IPFS Cluster v1.0.2 listening on:
        /ip4/192.168.1.3/tcp/9096/p2p/12D3KooWR9MhpEbn3GyzS1tS5cujgH4hqyDvsD5PPt6WATdJWKRy
        /ip4/127.0.0.1/tcp/9096/p2p/12D3KooWR9MhpEbn3GyzS1tS5cujgH4hqyDvsD5PPt6WATdJWKRy


2022-07-19T17:18:03.769Z	INFO	pinsvcapi	common/api.go:460	PINSVCAPI (HTTP): /ip4/127.0.0.1/tcp/9097
2022-07-19T17:18:03.769Z	INFO	restapi	common/api.go:460	RESTAPI (HTTP): /ip4/127.0.0.1/tcp/9094
2022-07-19T17:18:03.769Z	INFO	ipfsproxy	ipfsproxy/ipfsproxy.go:323	IPFS Proxy: /ip4/127.0.0.1/tcp/9095 -> /ip4/127.0.0.1/tcp/5001
2022-07-19T17:18:03.770Z	INFO	crdt	go-ds-crdt@v0.3.6/set.go:122	Tombstones have bloomed: 0 tombs. Took: 63.254µs
2022-07-19T17:18:03.770Z	INFO	crdt	go-ds-crdt@v0.3.6/crdt.go:296	crdt Datastore created. Number of heads: 0. Current max-height: 0. Dirty: false
2022-07-19T17:18:03.770Z	INFO	crdt	crdt/consensus.go:318	'trust all' mode enabled. Any peer in the cluster can modify the pinset.
2022-07-19T17:18:03.770Z	INFO	crdt	go-ds-crdt@v0.3.6/crdt.go:505	store is marked clean. No need to repair
2022-07-19T17:18:03.770Z	INFO	cluster	ipfs-cluster/cluster.go:733	Cluster Peers (without including ourselves):
2022-07-19T17:18:03.770Z	INFO	cluster	ipfs-cluster/cluster.go:735	    - No other peers
2022-07-19T17:18:03.770Z	INFO	cluster	ipfs-cluster/cluster.go:748	** IPFS Cluster is READY **
2022-07-19T17:18:03.771Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/id": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:817	Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	diskinfo	disk/disk.go:124	Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/pin/ls?stream=true&type=recursive": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:550	error querying pinset: %sPost "http://127.0.0.1:5001/api/v0/pin/ls?stream=true&type=recursive": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:03.771Z	ERROR	pintracker	stateless/stateless.go:667	Post "http://127.0.0.1:5001/api/v0/pin/ls?stream=true&type=recursive": dial tcp 127.0.0.1:5001: connect: connection refused


[nix-shell:~/.cache/nixpkgs-review/pr-181382]$ ./results/ipfs-cluster/bin/ipfs-cluster-ctl id
2022-07-19T17:18:16.324Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/id": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:16.324Z	INFO	restapilog	common/api.go:117	127.0.0.1 - - [19/Jul/2022:17:18:16 +0000] "GET /id HTTP/1.1" 200 591

12D3KooWR9MhpEbn3GyzS1tS5cujgH4hqyDvsD5PPt6WATdJWKRy | applePrincess | Sees 0 other peers
  > Addresses:
    - /ip4/127.0.0.1/tcp/9096/p2p/12D3KooWR9MhpEbn3GyzS1tS5cujgH4hqyDvsD5PPt6WATdJWKRy
    - /ip4/192.168.1.3/tcp/9096/p2p/12D3KooWR9MhpEbn3GyzS1tS5cujgH4hqyDvsD5PPt6WATdJWKRy
  > IPFS ERROR: Post "http://127.0.0.1:5001/api/v0/id": dial tcp 127.0.0.1:5001: connect: connection refused

[nix-shell:~/.cache/nixpkgs-review/pr-181382]$ 2022-07-19T17:18:18.770Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/id": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:18.772Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:640	error posting to IPFS:Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:18.772Z	ERROR	ipfshttp	ipfshttp/ipfshttp.go:817	Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
2022-07-19T17:18:18.772Z	ERROR	diskinfo	disk/disk.go:124	Post "http://127.0.0.1:5001/api/v0/repo/stat?size-only=true": dial tcp 127.0.0.1:5001: connect: connection refused
fg
./results/ipfs-cluster/bin/ipfs-cluster-service daemon
^C2022-07-19T17:18:23.550Z	INFO	cluster	ipfs-cluster/cluster.go:779	shutting down Cluster
2022-07-19T17:18:23.550Z	INFO	restapi	common/api.go:501	stopping Cluster API
2022-07-19T17:18:23.550Z	INFO	pinsvcapi	common/api.go:501	stopping Cluster API
2022-07-19T17:18:23.550Z	INFO	ipfsproxy	ipfsproxy/ipfsproxy.go:289	stopping IPFS Proxy
2022-07-19T17:18:23.550Z	INFO	crdt	crdt/consensus.go:348	stopping Consensus component
2022-07-19T17:18:23.550Z	INFO	monitor	pubsubmon/pubsubmon.go:192	stopping Monitor
2022-07-19T17:18:23.550Z	INFO	ipfshttp	ipfshttp/ipfshttp.go:252	stopping IPFS Connector
2022-07-19T17:18:23.550Z	INFO	pintracker	stateless/stateless.go:270	stopping StatelessPinTracker

[nix-shell:~/.cache/nixpkgs-review/pr-181382]$ ./results/ipfs-cluster/bin/ipfs-cluster-ctl --help
NAME:
   ipfs-cluster-ctl - CLI for IPFS Cluster

USAGE:
   ipfs-cluster-ctl [global options] command [command options] [arguments...]

VERSION:
   1.0.2

DESCRIPTION:

ipfs-cluster-ctl is a tool to manage IPFS Cluster nodes.

Use "ipfs-cluster-ctl help" to list all available commands and
"ipfs-cluster-ctl help <command>" to get usage information for a
specific one.

ipfs-cluster-ctl uses the IPFS Cluster API to perform requests and
display responses in a user-readable format. The location of the IPFS
Cluster server is assumed to be /ip4/127.0.0.1/tcp/9094, but can be configured
with the --host option. If several multiaddresses are specified
(comma-separated), requests will be sent to the first one and fail-over
over to the others. This also works for dns-based addresses which resolve
to multiple values.

To use the secure libp2p-http API endpoint, use "--host" with
the full cluster libp2p listener address, including the "/p2p/<peerID>"
part, or a /dnsaddr that resolves to it. Provide the cluster secret with
--secret as needed.

For feedback, bug reports or any additional information, visit
https://github.com/ipfs-cluster/ipfs-cluster.


COMMANDS:
   id       Retrieve peer information
   peers    List and manage IPFS Cluster peers
   add      Add a file or directory to ipfs and pin it in the cluster
   pin      Pin and unpin and list items in IPFS Cluster
   status   Retrieve the status of tracked items
   recover  Recover tracked items in error state
   version  Retrieve cluster version
   health   Cluster monitoring information
   ipfs     Manage IPFS daemon
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --host value, -l value         API endpoint multiaddresses (comma-separated) (default: "/ip4/127.0.0.1/tcp/9094")
   --secret value                 cluster secret (32 byte pnet-key) as needed. Only when using the LibP2P endpoint
   --https, -s                    use https to connect to the API
   --no-check-certificate         do not verify server TLS certificate. only valid with --https flag
   --encoding value, --enc value  output format encoding [text, json] (default: "text")
   --timeout value, -t value      number of seconds to wait before timing out a request (default: 0)
   --debug, -d                    set debug log level
   --basic-auth value             <username>[:<password>] specify BasicAuth credentials for server that
requires authorization. implies --https, which you can disable with --force-http [$CLUSTER_CREDENTIALS]
   --force-http, -f  force HTTP. only valid when using BasicAuth
   --help, -h        show help
   --version, -v     print the version
```

change looks good to me.

File: pkgs/applications/networking/ipfs-cluster/default.nix [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
 buildGoModule rec {
   pname = "ipfs-cluster";
-  version = "1.0.1";
+  version = "1.0.2";
 
-  vendorSha256 = "sha256-V+fqyrol+hXjjaCBlAs6f7FeqBqa2jTmMO2bvb6HfgY=";
+  vendorSha256 = "sha256-4pCJnQ/X5bvlgyHcRVZ8LyOexaKmz+1xAntMpZCpvd0=";
 
   src = fetchFromGitHub {
     owner = "ipfs";
```
Comment (bobby285271) [inline pkgs/applications/networking/ipfs-cluster/default.nix:10]: The repo is moved to https://github.com/ipfs-cluster/ipfs-cluster/, will be nice to update this as well.

Comment (Luflosi) [reply pkgs/applications/networking/ipfs-cluster/default.nix]: This wasn't yet the case, when I created the PR, so thank you for catching that.

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 181382` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ipfs-cluster</li>
  </ul>
</details>

**Comment by applePrincess** [comment]
Looks good for `x86_64-linux` , however tests failed on `x86_64-darwin` ran by ofborg. Could you re-check ?

**Comment by Luflosi** [comment]
Weird, I just checked out this branch, executed `nix-build -A ipfs-cluster` and the tests ran perfectly fine on my `x86_64-darwin` machine.
Perhaps the test is flakey? What else could cause this?

**Comment by applePrincess** [comment]
I don't have darwin so I cannot help here. 
Check by running `nixpkgs-review pr 181382` because that is effectively what is doing by ofborg. So you can inspect further.

**Comment by Luflosi** [comment]
During the tests, the macOS firewall asks three times if I wish to allow incoming connections. If I click the button to allow it or I just ignore the dialog box and do nothing, then the tests pass but if I click the button to deny incoming connections, the test fails with the same error message as OfBorg.
screenshot 1
screenshot 2
screenshot 3
Here is my translation of the text in the dialog box:

> Do you want the program "common.test" to accept incoming network connections?
> Clicking on "Do not allow" may limit the behaviour of the program. This setting can be changed in the system preferences "Security" under "Firewall".

If I deactivate the macOS firewall, the tests also pass.
There is also an option to block all incoming connections, which also causes the test to pass for some reason.

So in summary, I suspect that this may have something to do with the macOS firewall but I'm not sure what the difference between my machine and OfBorg is.

**Comment by applePrincess** [comment]
Yeah... really weired. Looking hydra job [log for ipfs-cluster 1.0.1](https://hydra.nixos.org/build/184801091/nixlog/1) this looks awfully similar to what is failing here. except the hydra one passes the test.

**Comment by Luflosi** [comment]
I think those errors in the log are expected. Compare it to the project's official CI run: https://github.com/ipfs-cluster/ipfs-cluster/runs/7218915278?check_suite_focus=true.

I think the error in this PR is something different. Should I just rebase the commit so the test runs again? Maybe it's just a flaky test. If so, I can report it upstream.

**Comment by applePrincess** [comment]
I don't want to compare upstream test result because they only run on ubuntu that will likely differ from what we see here on darwin. 
And that is why I was comparing with older version of hydra job which ran on darwin. 

Yeah agree to rebase this, if tests failing on ofborg but you can make sure that test still passes on your local machine then I can accept this.

**Review by Skin881kd — APPROVED** [review]

**Comment by bobby285271** [comment]
@ofborg build ipfs-cluster ipfs-cluster.passthru.tests

**Review by applePrincess — APPROVED** [review]
It was flakey indeed.
